TypeScript compilation fails with version v0.2.27 : error TS2611

**Issue type:**

<!--
     Have a question?
     Check the "Support" Documentation on the best places to ask questions!
     ---
     https://github.com/typeorm/typeorm/blob/master/docs/support.md
-->

[x] bug report
[ ] feature request
[ ] documentation issue

**TypeORM version:**

[x] `latest`
[x] `0.2.27` 

**Steps to reproduce or a small repository showing the problem:**

The issue appears to be related to `MongoEntityManager.d.ts`:

```shell
[node_modules/typeorm/entity-manager/MongoEntityManager.d.ts:27:9 - error TS2611: 'queryRunner' is defined as a property in class 'EntityManager', but is overridden here in 'MongoEntityManager' as an accessor.

27     get queryRunner(): MongoQueryRunner;
           ~~~~~~~~~~~

Found 1 error(s).

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I manually installed 0.2.26 and compilation succeeded.

--- a/package.json
+++ b/package.json
@@ -111,7 +111,7 @@
     "sqlite3": "^5.0.0",
     "ts-node": "^9.0.0",
     "typeorm-aurora-data-api-driver": "^1.4.0",
-    "typescript": "^3.9.7"
+    "typescript": "~3.6.0"
   },
   "dependencies": {
     "@sqltools/formatter": "1.2.2",

--- a/src/entity-manager/MongoEntityManager.ts
+++ b/src/entity-manager/MongoEntityManager.ts
@@ -570,7 +570,7 @@ export class MongoEntityManager extends EntityManager {
         if (!optionsOrConditions)
             return undefined;
 
-        if (FindOptionsUtils.isFindManyOptions(optionsOrConditions))
+        if (FindOptionsUtils.isFindManyOptions<Entity>(optionsOrConditions))
         // If where condition is passed as a string which contains sql we have to ignore
         // as mongo is not a sql database
             return typeof optionsOrConditions.where === "string"
@@ -587,7 +587,7 @@ export class MongoEntityManager extends EntityManager {
         if (!optionsOrConditions)
             return undefined;
 
-        if (FindOptionsUtils.isFindOneOptions(optionsOrConditions))
+        if (FindOptionsUtils.isFindOneOptions<Entity>(optionsOrConditions))
         // If where condition is passed as a string which contains sql we have to ignore
         // as mongo is not a sql database
             return typeof optionsOrConditions.where === "string"

--- a/src/find-options/FindOptionsUtils.ts
+++ b/src/find-options/FindOptionsUtils.ts
@@ -17,8 +17,8 @@ export class FindOptionsUtils {
     /**
      * Checks if given object is really instance of FindOneOptions interface.
      */
-    static isFindOneOptions(obj: any): obj is FindOneOptions<any> {
-        const possibleOptions: FindOneOptions<any> = obj;
+    static isFindOneOptions<Entity = any>(obj: any): obj is FindOneOptions<Entity> {
+        const possibleOptions: FindOneOptions<Entity> = obj;
         return possibleOptions &&
                 (
                     Array.isArray(possibleOptions.select) ||
@@ -41,8 +41,8 @@ export class FindOptionsUtils {
     /**
      * Checks if given object is really instance of FindManyOptions interface.
      */
-    static isFindManyOptions(obj: any): obj is FindManyOptions<any> {
-        const possibleOptions: FindManyOptions<any> = obj;
+    static isFindManyOptions<Entity = any>(obj: any): obj is FindManyOptions<Entity> {
+        const possibleOptions: FindManyOptions<Entity> = obj;
         return possibleOptions && (
             this.isFindOneOptions(possibleOptions) ||
             typeof (possibleOptions as FindManyOptions<any>).skip === "number" ||

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,5 +1,4 @@
 {
-    "version": "2.1.1",
     "compilerOptions": {
         "lib": ["es5", "es6"],
         "outDir": "build/compiled",
@@ -22,6 +21,11 @@
         "noUnusedLocals": true,
         "downlevelIteration": true
     },
+    "include": [
+      "sample",
+      "src",
+      "test"
+    ],
     "exclude": [
         "tmp",
         "temp",

